### DISCUSSION TITLE
Angular/NGRX 14 Support

### DISCUSSION THREAD

**Discussion by PointSingularity** [nilsmehlhorn/ngrx-wieder#64] [General]
First off, thanks for the great library!

I was wondering if there were plans for upgrading to Angular and NGRX version 14?

I also have a question about eslint-plugin-ngrx  and ngrx-wieder, now that the eslint-plugin officially is a part of the NGRX package:
When I use them in conjunction with each other, I get warnings about good action hygiene, specifically 
![image](https://user-images.githubusercontent.com/56246797/174735649-b3daad7d-5192-4ebc-ac95-7af7954ad429.png)

This error is easily solvable with:
```
// *.actions.ts
export const undo = createAction('UNDO');


// *.component.ts
...  
undo() {
    this.store.dispatch(undo());
  }
...
```

That creates another eslint warning, though:
![image](https://user-images.githubusercontent.com/56246797/174736197-ca462160-6353-45d8-8d4a-6005da40a065.png)

These errors can be probably be safely ignored/turned off, but I was just wondering if there was an alternative solution.

Thanks in advance!

**Comment by nilsmehlhorn** [comment]
Hey there!

Angular & NgRx 14 support will be coming up :)

Regarding the linting error: you can pass custom action types for undo/redo while [configuring](https://github.com/nilsmehlhorn/ngrx-wieder#configuration) ngrx-wieder:

```ts
// undo-redo.actions.ts
export const undo = createAction('[UI] Undo');
export const redo = createAction('[UI] Redo');

// *.reducer.ts
import { undo, redo } from 'undo-redo.actions'

const { createUndoRedoReducer } = undoRedo({
  undoActionType: undo.type,
  redoActionType: redo.type
});

// *.component.ts
undo() {
    this.store.dispatch(undo());
}
```

**Reply by PointSingularity** [reply]
Thanks for the quick reply!

Those are great news and thank you for the linting error solution, much appreciated 😃
